framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java

```diff
@@ -2952,15 +2952,15 @@ private void putOrAddToDeclAnnos(String key, Set<AnnotationMirror> annos) {
         if (stored == null) {
             annotationFileAnnos.declAnnos.put(key, new HashSet<>(annos));
         } else {
-            if (fileType != AnnotationFileType.JDK_STUB) {
-                stored.addAll(annos);
-            } else {
-                // JDK annotations should not replace any annotation of the same type.
-                List<AnnotationMirror> origStored = new ArrayList<>(stored);
-                for (AnnotationMirror anno : annos) {
-                    if (!AnnotationUtils.containsSameByName(origStored, anno)) {
-                        stored.add(anno);
-                    }
+            // JDK annotations should not replace any annotation of the same type.
+            List<AnnotationMirror> origStored = new ArrayList<>(stored);
```

Comment:
Why is this copy necessary? Is the point that if the `stored` contains `@Foo("A")`, no other `@Foo` can be added, but if `stored` contains no `@Foo` at all, `annos` can contain `@Foo("A") @Foo("B")` and all of them get added?
Isn't that rather odd logic?

Reply:
we can add a TODO for this odd behavior

Reply:
Left a comment about this issue.